### ISSUE TITLE
Invalid E0404 throw

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust#66736]
This code:

```rust
let x: i8 += 14;
```

gives the following error:

```text
error[E0404]: expected trait, found builtin type `i8`
 --> src/main.rs:5:12
  |
5 |     let x: i8 += 14;
  |            ^^ not a trait

warning: trait objects without an explicit `dyn` are deprecated
 --> src/main.rs:5:12
  |
5 |     let x: i8 += 14;
  |            ^^^^ help: use `dyn`: `dyn i8 +`
  |
  = note: `#[warn(bare_trait_objects)]` on by default

error: aborting due to previous error
```

It's not clear or even close to what the actual problem is. I think it should throw the E0067 error code.

This issue has been assigned to @mibac138 via [this comment](https://github.com/rust-lang/rust/issues/66736#issuecomment-624951014).
